### DISCUSSION TITLE
Service and Host Dependencies

### DISCUSSION THREAD

**Discussion by heinowalther** [sni/thruk#1305] [Q&A]
Hi there

A classic "mis-configuration" in nagios is to setup a lot of checks on a host, and as this host reboots or stops responding, you get bombarded with alerts....
I am seeking help to meditate this in the easiest way possible...
I am not use if the "parent" option helps with this, or if you have to use the Dependencies option.
But if you would like to use the Service Dependency, you have to create a dependency for each service?  As far as I can see there is no way to add this as an template when creating the service?
I have tested with a service dependency that "looks" at a "ping" check, and if this ping check is down, it will not check or notify for the specific service...  sadly you can only add one service check at a time... so quite a lot of extra work?

But maybe the parent option is the way to go here?  

Any suggestions are very welcome here :-)

/H

**Comment by sni** [comment]
Just a small note, service parents will be translated to service dependencies internally anyway, so it's just an easier way to create a dependency.
Another easy way is to use pattern matching and add a single dependency on each host which uses the service "*,!PING" and depends on "PING" or something like that. So you get a dependency for all services except the ping service.
